provider/storage_benchmark.py

```diff
@@ -0,0 +1,596 @@
+        """
+        self.params = params
+        self.session = session
+        self.test = test
+        super(Iozone, self).__init__(test, params, session, 'iozone')
+        self.vm = vm
+        self.tool_path = tool_path
+
+    def check_version(self, timeout=60):
+        """
+        check iozone version.
+        """
+        logging.info('Check the version of iozone.')
+        check_cmd = ' '.join((self.tool_path, '-v'))
+        results = self.session.cmd_output(cmd=check_cmd, timeout=timeout)
+        for info in results.splitlines():
+            if 'Version $Revision:' in info:
+                logging.info(info.strip())
+                break
+        else:
+            self.test.error('Failed to get the version of iozone.%s' % results)
+
+    def post_result(self):
+        """
+        Post result after iozone test, generate a series graphs.
+
+        """
+        # use this function, params "-i" for iozone is not suggested.
+        # suggest usage:iozone.exe -azR -r 64k -n 125M -g 512M -M -f
+        # I:\testfile". Otherwise, your result can't be analysed correctly.
+        logging.info("Generate graph of test result.")
+
+        random_string = utils_misc.generate_random_string(8)
+        stamp = time.strftime('%Y-%m-%d-%H-%M-%S') + random_string
+        analysis_name = '-'.join(('iozone', 'analysis', 'results', stamp))
+        analysis_path = os.path.join(self.test.resultsdir, analysis_name)
+
+        iozone_analyzer = postprocess_iozone.IOzoneAnalyzer(
+            list_files=[self.ret_path], output_dir=analysis_path)
+        iozone_analyzer.analyze()
+        iozone_plotter = postprocess_iozone.IOzonePlotter(
+            results_file=self.ret_path, output_dir=analysis_path)
+        iozone_plotter.plot_all()
+
+    def run(self, cmd_options='-a', record=True, post=False,
+            timeout=1800, debug=True):
+        """
+        Run iozone test in guest.
+
+        :param cmd_options: iozone command options.
+        e.g: -azR -r 64k -n 1G -g 1G -M -b iozone.xls -f E:\testfile.
+        :param record: bool, if is True, will record the test results,
+        otherwise do not.
+        :param post: bool, if is True, will post the test results,
+        otherwise do not.
+        :param timeout: timeout
+        :param debug: bool, if is True, print the output of testing.
+        """
+        cmd = ' '.join((self.tool_path, cmd_options))
+        output = super(Iozone, self).run(cmd, timeout, record, debug)
+        if post and record:
+            if not re.search(r'\s+-i\s+\S+', cmd_options):
+                self.post_result()
+            else:
+                logging.warn('Maybe not be analysed correctly '
+                             'since iozone commands includes \'-i\' '
+                             'option. So Cancel it')
+        return output
+
+
+class IozoneLinux(Iozone):
+    def __init__(self, test, params, vm, session):
+        self.session = vm_session_check(vm, session)
```

Comment:
I think the session's state is better maintained outside `__init__`, that is too say when people trying to use these class a living session should be provide.